odoo/tools/barcode.py

```diff
@@ -52,3 +53,9 @@ def check_barcode_encoding(barcode, encoding):
            and len(barcode) == barcode_size \
            and re.match(r"^\d+$", barcode) \
            and get_barcode_check_digit(barcode) == int(barcode[-1])
+
+def check_barcode_is_code128(barcode):
+    unsupported_chars = re.findall("[^\x00-\x7F\xF1-\xF4]", barcode)
+    if barcode and unsupported_chars:
+        error_message = _("Invalid Barcode Format, following characters are not supported: %s")
+        raise ValidationError(error_message % " ".join(unsupported_chars))
```

Comment:
```suggestion
    if barcode:
        unsupported_chars = re.findall("[^\x00-\x7F\xF1-\xF4]", barcode)
        if unsupported_chars:
            error_message = _("Invalid Barcode Format, following characters are not supported: %s")
            raise ValidationError(error_message % " ".join(unsupported_chars))
```

Reply:
```suggestion
    if barcode:
        unsupported_chars = re.findall("[^\x00-\x7F\xF1-\xF4]", barcode)
        if unsupported_chars:
            error_message = _(
                "Invalid Barcode Format, following characters are not supported: %s",
                " ".join(unsupported_chars))
            raise ValidationError(error_message)
```
to use the right translation API (with fallback to eng if the translated string doesn't contain the expected %s)

Reply:
Hello @svs-odoo @Feyensv 
Changes have been done, kindly review them.
Thanks.